File: web/src/models/Author.ts
/**
 * Author of a book
 * @export
 * @interface Author
 */
export default interface Author {
    /**
     * Internal author identifier
     * @type {string}
     * @memberof Author
     */
    id: string;
    /**
     * Internal author identifier
     * @type {string}
     * @memberof Author
     */
    _id: string;
    /**
     * Author's last name
     * @type {string}
     * @memberof Author
     */
    last_name?: string;
    /**
     * Author's first name
     * @type {string}
     * @memberof Author
     */
    first_name?: string;
    /**
     * Author's middle name
     * @type {string}
     * @memberof Author
     */
    middle_name?: string;
}

File: dto/author.py
class Author(object):

    def __init__(self,
                 id: str = None,
                 first_name: str = None,
                 last_name: str = None,
                 middle_name: str = None
                 ):
        self.id = id
        self.first_name = first_name
        self.last_name = last_name
        self.middle_name = middle_name

    @property
    def name(self):
        result = '{} {} {}'.format(self.last_name, self.first_name, self.middle_name)
        return result
